reana/reana_dev/release.py

```diff
@@ -202,4 +249,90 @@ def release_pypi(ctx, component, timeout):  # noqa: D301
+    :param components: The option ``component`` can be repeated. The value may
+                       consist of:
+                         * (1) standard component name such as
+                               'reana-workflow-controller';
+                         * (2) short component name such as 'r-w-controller';
+                         * (3) special value '.' indicating component of the
+                               current working directory;
+                         * (4) special value 'CLUSTER' that will expand to
+                               cover all REANA cluster components [default];
+                         * (5) special value 'CLIENT' that will expand to
+                               cover all REANA client components;
+                         * (6) special value 'DEMO' that will expand
+                               to include several runable REANA demo examples;
+                         * (7) special value 'ALL' that will expand to include
+                               all REANA repositories.
+    :type component: str
+    """
+
+    def _update_values_yaml(new_docker_images):
+        """Update all images in ``values.yaml``, skipping the ones up to date."""
+        update_images_cmds = []  # FIXME
+        values_yaml_relative_path = "helm/reana/values.yaml"
+        values_yaml_abs_path = os.path.join(
+            get_srcdir("reana"), values_yaml_relative_path
+        )
+        values_yaml = ""
+
+        with open(values_yaml_abs_path) as f:
+            values_yaml = f.read()
+            for docker_image in new_docker_images:
+                image_name, _ = docker_image.split(":")
+                if image_name in values_yaml:
+                    values_yaml = re.sub(
+                        f"{image_name}:.*", lambda _: docker_image, values_yaml, count=1
+                    )
+
+        with open(values_yaml_abs_path, "w") as f:
+            f.write(values_yaml)
+
+    remaining_docker_releases = []
+    new_docker_images = []
+    for component in REPO_LIST_CLUSTER:
+        if not is_component_dockerised(component):
+            continue
+        if not git_is_current_version_tagged(component):
+            remaining_docker_releases.append(component)
+        else:
+            new_docker_images.append(
+                f"{user}/{component}:{get_component_version_as_semver2(component)}"
+            )
+
+    if remaining_docker_releases:
+        line_by_line_missing_releases = "\n".join(remaining_docker_releases)
+        click.secho(
+            "The following components are missing to be released:\n"
+            f"{line_by_line_missing_releases}",
+            fg="red",
+        )
+        sys.exit(1)
+
+    _update_values_yaml(new_docker_images)
+    version = bump_component_version(
+        "reana",
+        get_current_component_version_from_source_files("reana"),
+        next_version=version,
+    )
+
+    if publish:
+        package_and_publish_helm_chart(user, version, component)
```

Comment:
This is what I've ended up with, `reana-dev release-helm` would do too many things.

I think we could split it as:
- Until line `330` (updating `values.yaml` and bumping version, similar to `reana-dev git-upgrade-shared-modules` but for Helm and docker images) we could have a new command e.g. `reana-dev helm-upgrade-components` or similar
- Then one would run `reana-dev git-create-release-commit`
- Lastly, the true `reana-dev release-helm` would just consist of `package_and_publish_helm_chart`

What do you think?